### ISSUE TITLE
Error: SQLite Expression tree is too large

### ISSUE THREAD

**Issue by JustAndrei** [Temptin/SkypeExport#11]
Hi!

I tried to run the Windows version and got the following error:

> skypeexport -t 24h
> Skype History Exporter v1.3.0 Stable
>    WEBSITE: [ https://github.com/Temptin/SkypeExport ]
> Error while processing Skype database: "Expression tree is too large (maximum depth 1000)".

**Comment by bofh** [comment]
The same on Mac OS X. v1.3.0 Stable

**Comment by Temptin** [comment]
I've never seen that error before. I read up on it and it's something SQLite says if you've got a query with way too many arguments within a single set of parenthesis.

Here's a stackoverflow example I found, describing what causes the bug:

```
Bad:
id = 387 OR id = 388 OR id = 390 OR id = 391 [...] OR id = 392

Bad:
(id = 387 OR id = 388 OR id = 390 OR id = 391 [...] OR id = 392)

Good:
(id = 387 OR id = 388 [...] OR id = 390) OR (id = 391 [...] OR id = 392) OR (...
```

So, checking the source for places that build long expressions, I think I found the culprit. This is the only place that has the potential of building a long SELECT statement:
https://github.com/Temptin/SkypeExport/blob/97dc2c2f855999e98a6b522f5baf8d737ff9d54e/src/SkypeExport/model/skypeparser_core.cpp#L139

Apparently you guys have both had thousands of multi-person conferences + participant combinations. That's what it sounds like.

I'll restructure that statement in a minor 1.4 update out soon, where I will also take care of #7. I'm setting up my build environment to take care of it now and that'll be the last official update. After that people are free to fork SkypeExport.

The new release will be available within a few hours at:
https://github.com/Temptin/SkypeExport/releases
